app/controllers/users_controller.rb

```diff
@@ -6,11 +6,25 @@ def new
   end
 
   def create
+
+    all_users = User.all.map { |x| x.email }
```

Comment:
This is a more expensive and complicated query than we'd like to make here and it becomes more expensive the more users sign up for the site since we have to process *all* of them.

What would be better is something like:
```
@user = User.find_by(email: params[:email])
if @user
  flash[:blah] = asldjflkasjdf
  render :new
else
  @user = User.find_or_create_by( lkjasldkjfas)
  flash[:notice] = "cool~!"
  redirect_to :bar
end
```